fe/fe-core/src/main/java/com/starrocks/consistency/MetaRecoveryDaemon.java

```diff
@@ -0,0 +1,347 @@
+                                                table.getId(), table.getName(), database.getId(), database.getFullName());
+                                    }
+
+                                    Set<Long> replicaVersions = localTablet.getAllReplicaVersions();
+                                    if (isFirstTablet) {
+                                        for (Long version : replicaVersions) {
+                                            commonVersion.put(version, true);
+                                        }
+                                        isFirstTablet = false;
+                                    } else {
+                                        for (Map.Entry<Long, Boolean> entry : commonVersion.entrySet()) {
+                                            if (!replicaVersions.contains(entry.getKey())) {
+                                                entry.setValue(false);
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            if (foundCommonVersion && recoverVersion != physicalPartition.getVisibleVersion()) {
+                                if (GlobalStateMgr.getCurrentState().getGlobalTransactionMgr()
+                                        .hasCommittedTxnOnPartition(dbId, table.getId(), physicalPartition.getId())) {
+                                    LOG.warn("There are committed txns waiting to publish, " +
+                                                    "ignore partition: {}-{} table: {}-{} db: {}-{}",
+                                            physicalPartition.getId(), partition.getName(), table.getId(),
+                                            table.getName(), database.getId(), database.getFullName());
+                                    UnRecoveredPartition recoveryInfo = new UnRecoveredPartition(database.getFullName(),
+                                            table.getName(), partition.getName(), physicalPartition.getId(),
+                                            "There are committed txns waiting to publish, ignore this partition");
+                                    addToUnRecoveredPartitions(recoveryInfo);
+                                } else {
+                                    partitionsToRecover.add(new PartitionVersion(dbId, table.getId(),
+                                            physicalPartition.getId(), recoverVersion));
+                                }
+                            } else if (!foundCommonVersion) {
+                                long maxCommonVersion = -1L;
+                                for (Map.Entry<Long, Boolean> entry : commonVersion.entrySet()) {
+                                    if (entry.getValue() && entry.getKey() > maxCommonVersion) {
+                                        maxCommonVersion = entry.getKey();
```

Comment:
what if it doesn't have common version? like 2 bucket, replication_num = 3,  tablet1(2,2,2), tablet2(3,3,4)

Reply:
Not covered this case